src/base/ct2ctml.cpp

```diff
@@ -85,6 +89,52 @@ static std::string call_ctml_writer(const std::string& text, bool isfile)
     } else {
         file = "<string>";
         arg = "text=r'''" + text + "'''";
+
+    }
+
+    // If the user wants to convert a mechanism using a text passed via the
+    //       source="""..."""
+    // argument in python, then we have to make sure that it is short enough
+    // to fit in the command line when routed to python as:
+    //       python -c ...
+    // statement downstream in the code
+
+    // So, check the max size of a string that can be passed on the command line
+    // This is OS Specific. *nix systems have the sysconf() function that tells 
+    // us the largest argument we can pass. Since such a function does not exist
+    // for Windows, we set an arbitrary limit of 80,000 chars
+#ifdef _WIN32
+    int max_argv_size = 80000;
+#endif
+
+#ifndef _WIN32
+    int max_argv_size = sysconf(_SC_ARG_MAX);
+#endif
+
+    if (text.size() > max_argv_size) {
```

Comment:
`max_argv_size` is the length of the whole of `argv`, so we should add the approximate length of the rest of the command (~500 characters) here.